### DISCUSSION TITLE
Adding a move list function

### DISCUSSION THREAD

**Discussion by thewhisperinyourears** [DOBRO/binbo#14] [Ideas]
is there a way we can add a move list? I know sometimes we will not have a move history if the game is loaded by fen. But I find myself needing to add move_history or move_list however you want to call it.

**Comment by DOBRO** [comment]
Could you explain how would you use move history?

**Comment by thewhisperinyourears** [comment]
Normally I would use a move history to display it in the client. Otherwise I would need to manually have a move list object. I would rather have it in the game. Just like chess.js does. We can have a move history for the moves we have entered or otherwise if the game has no moves because it was loaded by FEN, then just show an empty [] or show the moves. We need to know how to show the moves because in chess normally is a pair of moves so it makes sense to have it as [{e4, e5}] or maybe a plain list [e4, e5]. I think the latter since better no? I just want to sync the fact that a position may be loaded by FEN and is black to move so the move would be like [ e5 ] obviously we are missing white move e4 so the way to show it in the front end can be confusing.

**Comment by DOBRO** [comment]
Ok, let me remind you that the binbo's game server (gen_server) is a temporary Erlang process itself. What would you do if it's crashed or just stopped somehow? How would you restore the move history? No way!
So keeping history in a database is the only right choice.
I would suggest you a better way to save the history with another approach. As soon as the move is done, save it immediately or add it to your own move list. If you don't need a database, save it in the process which sends moves (messages actually) to the game process.

BTW, binbo does know nothing which move notation type you want to use, SAN (e4) or square notation (e2-e4, e2e4, {e2, e4}, etc...). It's up to you.
